pyrealm/core/solar.py

```diff
@@ -1,26 +1,600 @@
-"""The ``solar`` submodule provides functions and classes to calculate daily solar
-radiation fluxes and other radiative values.
+"""The :mod:`~pyrealm.core.solar` submodule provides functions to calculate
+photosynthetic photon flux density (ppfd), daily solar radiation fluxes and other
+radiative values.
 """  # noqa: D205
 
 import numpy as np
 from numpy.typing import NDArray
 
 from pyrealm.constants import CoreConst
+from pyrealm.core.calendar import LocationDateTime
+from pyrealm.core.utilities import check_input_shapes
 
-# from pyrealm.splash.const import (
-#     kA,
-#     kalb_sw,
-#     kalb_vis,
-#     kb,
-#     kc,
-#     kd,
-#     ke,
-#     keps,
-#     kfFEC,
-#     kGsc,
-#     komega,
-#     pir,
-# )
+
+def calc_distance_factor(nu: NDArray, k_e: float) -> NDArray:
+    r"""Calculates distance factor.
+
+    This function calculates the distance factor :math:`dr` using the method of
+    :cite:t:`berger:1993a`.
+
+    .. math::
+
+        dr = \left( 1.0 \mathbin{/}
+               \left(\frac{1.0 - k_e^2}
+                          {1.0 + k_e \cos\left(\nu \cdot \pi \mathbin{/} 180)\right)}
+               \right)
+             \right)^2
+
+    Args:
+        nu          : Heliocentric true anomaly (degrees)
+        k_e         : Solar eccentricity
+
+    Returns:
+        An array of distance factors
+    """
+
+    dr = (1.0 / ((1.0 - k_e**2) / (1.0 + k_e * np.cos(np.deg2rad(nu))))) ** 2
+
+    return dr
+
+
+def calc_declination_angle_delta(
+    lambda_: NDArray, k_eps: float, k_pir: float
+) -> NDArray:
+    r"""Calculates declination angle delta.
+
+    This function calculates the solar declination angle delta using
+    the method of :cite:t:'Woolf:1968'.
```

Comment:
```suggestion
    This function calculates the solar declination angle delta using the method of 
    :cite:t:`Woolf:1968`.
```

Two things there, one of which is hugely petty 😈 :
* The citation syntax needs backticks (here and elsewhere).
* The docstrings could use more of the linewidth - the project uses 88 characters. I use the Rewrap plugin for VSCode, which is great for wrapping and re-wrapping strings and comments: https://marketplace.visualstudio.com/items?itemName=stkb.rewrap

Reply:
Have sorted this now